apps/ejabberd/src/ejabberd_binary.erl

```diff
@@ -28,3 +29,16 @@ string_to_binary(S) when is_list(S) ->
     end;
 string_to_binary(B) when is_binary(B) ->
     B.
+
+
+join([], _) ->
```

Comment:
Actually some time ago I saw some micro-benchmarking done by @arcusfelis showing that `io_list_to_binary` is faster than incrementally built binary.